catkin_ws/src/planner/src/substates/utility/controller.py

```diff
@@ -169,35 +169,35 @@ def rotateEuler(self,ang):
         if z is None: z = self.theta_z
         self.rotate(euler_to_quaternion(x,y,z))
 
-    def state(self,pos,ang):
-        x,y,z = pos
-        if any(x is None for x in ang) and any(x is not None for x in ang):
-            raise ValueError("Invalid state goal: quaternion cannot have a combination of None and valid values. Goal received: {}".format(ang))
-        w,wx,wy,wz = ang
-        goal_state = self.get_state_goal([x,y,z,w,wx,wy,wz],do_not_displace)
-        self.StateQuaternionStateClient.send_goal_and_wait(goal_state)
-    
-    def stateDelta(self,pos,ang):
+
+    def state(self, pos, ang, isEuler=False, isDelta=False):
+        if isEuler:
+            wx,wy,wz = ang
+
+            if not isDelta:
```

Comment:
when doing a delta control, you do that by passing do_displace to the function that gives you the goal. look at the other delta functions for inspiration. I don't think this if is necessary